common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java

```diff
@@ -70,6 +70,8 @@ public class TransportResponseHandler extends MessageHandler<ResponseMessage> {
   private static ScheduledExecutorService fetchTimeoutChecker = null;
   private ScheduledFuture fetchCheckerScheduleFuture;
 
+  private final Object handleFailureLock = new Object();
```

Comment:
IMO we don't need the lock here, because `outstandingFetches/outstandingRpcs/outstandingPushes` are `ConcurrentHashMap`s, they can guarantee `remove(requestId)` is atomic. Except for `failOutstandingRequests`, the `callback` will only be called after it is removed from the `ConcurrentHashMap`, so we only need to consider refactoring `failOutstandingRequests`. IMO, we can loop over the key iterator, inside each loop we call `remove(key)`, if the returned value is not null, call `onFailure` on it.

Reply:
Now outstandingFetches/outstandingRpcs is still Map. Change them to ConcurrentHashMap and `failOutstandingRequests` handle requests one by one seems to work. I will have a try later.

Reply:
@onebox-li They are already `ConcurrentHashMap`, see
```
    this.outstandingFetches = JavaUtils.newConcurrentHashMap();
    this.outstandingRpcs = JavaUtils.newConcurrentHashMap();
    this.outstandingPushes = JavaUtils.newConcurrentHashMap();
```

Reply:
Thanks, got it.